care/facility/api/serializers/post_covid_data.py

```diff
@@ -0,0 +1,175 @@
+    pulse_rate = serializers.IntegerField(allow_null=True)
+    blood_pressure_systolic = serializers.IntegerField(allow_null=True)
+    blood_pressure_diastolic = serializers.IntegerField(allow_null=True)
+    respiration_rate = serializers.IntegerField(allow_null=True)
+    temperature = serializers.FloatField(allow_null=True)
+    oxygen_saturation = serializers.IntegerField(allow_null=True)
+
+
+# Validator serializer for each object of steroids_drugs,
+# antibiotics_drugs and antifungals_drugs fields
+class DrugInfoSerializer(serializers.Serializer):
+    name = serializers.CharField()
+    duration = serializers.IntegerField()
+
+
+# validator serializer for anticoagulants_drugs field
+class AnticoagulantDrugInfoSerializer(serializers.Serializer):
+    mode_of_transmission = ChoiceField(choices=ModeOfTransmissionChoices)
+    name = serializers.CharField()
+    duration = serializers.IntegerField()
+
+
+# validator serializer for systemic_examination field
+class SystemicExaminationSerializer(serializers.Serializer):
+    respiratory = serializers.JSONField()
+    cvs = serializers.CharField(style={'base_template': 'textarea.html'}, allow_null=True, allow_blank=True)
+    cns = serializers.CharField(style={'base_template': 'textarea.html'}, allow_null=True, allow_blank=True)
+    git = serializers.CharField(style={'base_template': 'textarea.html'}, allow_null=True, allow_blank=True)
+
+    def validate_respiratory(self, respiratory):
+        if (
+            len(respiratory) != 3 or
+            not RespiratorySerializer(data=respiratory).is_valid()
+        ):
+            raise serializers.ValidationError(
+                "Respiratory part of systemic examination is invalid"
+            )
+
+
+# validator serializer for respiratory attribute of systemic_examination field
+class RespiratorySerializer(serializers.Serializer):
+    wob = serializers.CharField(style={'base_template': 'textarea.html'}, allow_null=True, allow_blank=True)
+    rhonchi = serializers.CharField(style={'base_template': 'textarea.html'}, allow_null=True, allow_blank=True)
+    crepitation = serializers.CharField(style={'base_template': 'textarea.html'}, allow_null=True, allow_blank=True)
+
+
+class PostCovidDataSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PostCovidData
+        read_only_fields = TIMESTAMP_FIELDS
+        exclude = TIMESTAMP_FIELDS + ("id",)
+        required_fields = [
+            "post_covid_time",
+            "date_of_test_positive",
+            "date_of_test_negative",
+            "treatment_facility",
+            "treatment_duration",
+            "icu_admission",
+            "oxygen_requirement",
+            "probable_diagnosis"
+        ]
+        extra_kwargs = {i: {'required': True} for i in required_fields}
+
+    def validate_vitals_at_admission(self, vitals_at_admission):
+        if vitals_at_admission == None:
+            return vitals_at_admission
+
```

Comment:
You may probably replace this with:
```suggestion
        if vitals_at_admission is not None:
```
and similarly for other validators as well.

Reply:
done